### PR TITLE
[Dynamic selection/pstl-offload] Add interception of dynamic_selection header

### PR DESCRIPTION
**Pull request by kboyarinov**
Add new `dynamic_selection` public header into pstl-offload interception mechanism + test. 

If `dynamic_selection` would be merged before `pstl-offload` - this PR should be moved into `pstl-offload`.
Otherwise, it should be added into `dynamic_selection`.

### CODE REVIEW

File: include/pstl_offload/oneapi/dpl/dynamic_selection [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_PSTL_OFFLOAD_TOP_LEVEL_ONEAPI_DPL_DYNAMIC_SELECTION
+
+#ifndef _ONEDPL_PSTL_OFFLOAD_TOP_LEVEL
+#    define _ONEDPL_PSTL_OFFLOAD_TOP_LEVEL_ONEAPI_DPL_DYNAMIC_SELECTION
+#    define _ONEDPL_PSTL_OFFLOAD_TOP_LEVEL
+#endif // _ONEDPL_PSTL_OFFLOAD_TOP_LEVEL
+
+#include_next <oneapi/dpl/dynamic_selection>
```
Comment (rarutyun) [inline include/pstl_offload/oneapi/dpl/dynamic_selection:17]: Have you checked it with script? if not do it, please.

File: test/pstl_offload/headers/headers.dpl/oneapi_dpl_dynamic_selection.pass.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <oneapi/dpl/dynamic_selection>
+#include "support/utils.h"
+
+int main() {
+    using namespace oneapi::dpl::experimental;
+    [[maybe_unused]] std::size_t r = sizeof(fixed_resource_policy<sycl_policy>);
```
Comment (rarutyun) [inline test/pstl_offload/headers/headers.dpl/oneapi_dpl_dynamic_selection.pass.cpp:15]: I don't remember `sycl_policy` API in dynamic selection. Are you sure this test passes?

File: include/pstl_offload/oneapi/dpl/dynamic_selection (hunk 1 above)
Comment (kboyarinov) [reply include/pstl_offload/oneapi/dpl/dynamic_selection:17]: The script did not report any issues.

File: test/pstl_offload/headers/headers.dpl/oneapi_dpl_dynamic_selection.pass.cpp (hunk 2 above)
Comment (kboyarinov) [reply test/pstl_offload/headers/headers.dpl/oneapi_dpl_dynamic_selection.pass.cpp]: Fixed, should be `sycl_backend`

**Review by rarutyun — APPROVED** [review]

**Pull request merged**